### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by IgorBochkor**
added solution

### CODE REVIEW

File: src/main/java/com/dev/boot/bankservice/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Override
+    public User get(Long id) {
+        return userRepository.findById(id).get();
+    }
+
+    @Override
+    public User update(User user) {
+        return userRepository.save(user);
+    }
```
Comment (yurwing) [inline src/main/java/com/dev/boot/bankservice/service/impl/UserServiceImpl.java]: checklist
instead of methods `create `and `update`, create method `save`

**Review by yurwing — APPROVED** [review]

File: src/main/java/com/dev/boot/bankservice/model/User.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToMany;
+import lombok.Data;
+
+@Entity
+@Data
+public class User {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/boot/bankservice/model/User.java:17]: add `@Table` annotation

File: src/main/java/com/dev/boot/bankservice/model/User.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String name;
+    @Column(name = "date_of_birth")
+    private LocalDate dateOfBirth;
+    @Column(unique = true, name = "phone_number")
+    private String phoneNumber;
+    private String password;
+    @ManyToMany(fetch = FetchType.EAGER)
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/boot/bankservice/model/User.java]: adding `FetchType.EAGER` is bad practice

File: src/main/java/com/dev/boot/bankservice/model/User.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    private Set<Role> roles;
+}
+
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/boot/bankservice/model/User.java]: remove redundant empty lines

File: src/main/java/com/dev/boot/bankservice/repository/UserRepository.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package com.dev.boot.bankservice.repository;
+
+import com.dev.boot.bankservice.model.User;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+    Optional<User> findByPhoneNumber(String phoneNumber);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/boot/bankservice/repository/UserRepository.java:10]: let's use join fetch in this method and in `findById()` as well

File: src/main/java/com/dev/boot/bankservice/service/UserService.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package com.dev.boot.bankservice.service;
+
+import com.dev.boot.bankservice.model.User;
+
+public interface UserService {
+    User create(User user);
+
+    User get(Long id);
+
+    User update(User user);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/boot/bankservice/service/UserService.java]: let's see checklist

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -1 +1 @@
-
+spring.datasource.url=jdbc:h2:mem:testdb
```
Comment (KseniiaMakarova) [inline src/main/resources/application.properties]: let's enable the console so that we could have access to our db from the browser

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**